lib/OpenQA/Utils.pm

```diff
@@ -1184,18 +1185,28 @@ sub rand_range { $_[0] + rand($_[1] - $_[0]) }
 sub in_range   { $_[0] >= $_[1] && $_[0] <= $_[2] ? 1 : 0 }
 
 sub set_listen_address {
-    my ($port) = @_;
-
-    if ($ENV{MOJO_LISTEN}) {
-        return;
-    }
+    my $port = shift;
 
+    return if $ENV{MOJO_LISTEN};
     my @listen_addresses = ("http://127.0.0.1:$port");
-    if (IO::Socket::IP->new(Listen => 5, LocalAddr => '::1')) {
-        push(@listen_addresses, "http://[::1]:$port");
-    }
 
-    $ENV{MOJO_LISTEN} = join(',', @listen_addresses);
+    # Check for IPv6
+    push @listen_addresses, "http://[::1]:$port" if IO::Socket::IP->new(Listen => 5, LocalAddr => '::1');
+
+    $ENV{MOJO_LISTEN} = join ',', @listen_addresses;
```

Comment:
I'm wondering why you prefer parenthesesless syntax when passing multiple arguments. It looks not very clean too me. Maybe because I'm also used to C/C++ where this looks like only the `','` is passed to `join` and `, @listen_addresses` has no effect at all.

E.g. in the following C code `a` becomes 4 or whatever the size of `int` is and `b` remains 42:
```
int a, b = 42;
a = sizeof a, b;
```

That's just a comment, I don't care that much.

Reply:
Because i see `join` as a Perl keyword, not a function. Both forms are generally considered clean Perl.

Reply:
I don't know how you've set up syntax highlighting for Perl, but i actually have keywords set to a different color than functions. So treating them differently has become a habit for me. For `join` is doesn't matter at all, but some keywords like `map` have special parsing rules that are different from what normal functions can do.

Reply:
I understand. But I never thought of join as a keyword - it is also listed as a function in the official docs: https://perldoc.perl.org/functions/join.html

I usually just follow the rule to use parenthesesless syntax only for one argument.

Reply:
It's the same for `return` 😉. https://perldoc.perl.org/functions/return.html